File: scraper-go/domain/rent/rent.go
package rent

import "fmt"

type Rent struct {
	Price, CommonServiceFee uint
}

func NewRent(price, commonServiceFee uint) *Rent {
	return &Rent{price, commonServiceFee}
}

func NewRents(prices, commonServiceFees []uint) ([]*Rent, error) {
	priceLen, commonServiceFeeLen := len(prices), len(commonServiceFees)
	if priceLen == 0 || commonServiceFeeLen == 0 {
		return nil, fmt.Errorf("not found arguments. price: %d, CommonServiceFee: %d", priceLen, commonServiceFeeLen)
	}
	if priceLen != commonServiceFeeLen {
		return nil, fmt.Errorf("incorrect size. price: %d, CommonServiceFee: %d", priceLen, commonServiceFeeLen)
	}
	var rents []*Rent
	for i := 0; i < priceLen; i++ {
		rents = append(rents, NewRent(prices[i], commonServiceFees[i]))
	}
	return rents, nil
}

File: scraper/src/core/entities/rent.ts
import { StringNumber } from './stringNumber';

type Rent = {
  price: number;
  commonServiceFee: number;
};
type CreateNewRent = (
  price: StringNumber,
  commonServiceFee: StringNumber,
) => Rent;
/**
 * @param price
 * @param commonServiceFee
 */
export const createNewRent: CreateNewRent = (price, commonServiceFee) => ({
  price: Number(price),
  commonServiceFee: Number(commonServiceFee),
});
